operatorcourier/const.py

```diff
@@ -0,0 +1,97 @@
+general_required_fields = [
+    "spec"
+]
+
+metadata_required_fields = [
+    {
+        "field" : "annotations",
+        "description" : "Without this field, your operator will be missing essential data for the UI. Please add this field to the CSV and run validation again to see what subfields are required.",
+        "required" : True
+    }
+]
+
+metadata_annotations_required_fields = [
+    {
+        "field" : "description",
+        "description" : "Without this field, the description displayed in the tiles of the UI will be a truncated version of spec.description.",
+        "required" : True
+    },
+    {
+        "field" : "categories",
+        "description" : "Without this field, the operator will be categorized as Other.",
```

Comment:
Since there is a default, should this be required? What factors decide which fields are "required"?

Reply:
Your (everyone added as a reviewer) feedback in this PR determines which fields are required

Reply:
Let's have this conversation before I review. In my opinion, `required` in this case should be limited to issues that would prevent the UI from displaying as intended. Anything that has a default should not be `required`.

Reply:
The thing is we have defaults out of necessity due to there being no enforcement on the csv for ui fields. Because operatorhub.io is in production we have had to implement fallbacks for every scenario because we can't have the UI failing. I agree we should have a discussion about what is required but it should not be on the basis of what would make the UI fail and should rather focus on what fields are required for the user experience to make sense.